### ISSUE TITLE
How long does the training take?

### ISSUE THREAD

**Issue by dongzizhu** [Holipori/EKAID#8]
Hi,

I'm trying to train the model for difference questions only. The only change I made to the code is turning the "simple" in transform_questions2dataset() of dataset_preparation to "True". However, it's taking a really long time to train and seems having a hard time converging. I'm training with one single GPU and it's been 24 hours for more than 17 epochs. How long does this usually take? Because when I trained for all of the questions, the training stopped after 4 epochs and reached a pretty good performance. Do you have any insight or suggestions regarding this?

<img width="653" alt="training" src="https://github.com/Holipori/EKAID/assets/38762574/1fd62e76-41c2-429f-a227-cbaf7f9fb98f">

Thx!

**Comment by Holipori** [comment]
Hi,

For evaluating the difference questions only. We filtered the output from the original output(trained on all questions) and kept the "difference" questions only. Then we run the evaluation code to obtain the results. 
I just updated the 'evaluate_score.py' file. To use the code for evaluating difference questions only, please use the command 
`python evaluate_score.py -n <run_name> -c <checkpoint_num> -t difference`

**Comment by dongzizhu** [comment]
Hi, 

thx for getting back to me. I'm not trying to just evaluate the model (trained with every questiobs) on difference question. I want to train it only with the difference questions. 

also, you didn't answer my question about the training time:)

**Comment by Holipori** [comment]
Hi,

The difference questions need to be trained based on stable performance on single-image problems. Without enough training on single-image questions, training only difference questions may not generate good enough results. That's why we train all questions together.

If you insist on training difference questions only, I would suggest exploring the best learning rate and other hyperparameters to get the best results. However, I still don't think that will surpass the performance of training all questions and then evaluating the difference questions. The training time should be similar to the original setting, which is around 4 hours for 40k iterations (two 3090 GPUs in our case).

Happy Holidays!
